Add guides for creating/maintaining a fork of a package that has moved into atom/atom

jasonrudolph opened the pull request:
### Background

In an effort to streamline the development of Atom by reducing development overhead, we're [consolidating core Atom packages](https://github.com/atom/atom/blob/051bfe5d22d209e8fb4a1d91fe2a9251c97f370c/docs/rfcs/003-consolidate-core-packages.md) into the [atom/atom repository](https://github.com/atom/atom). In some cases, package authors have used these core packages as a starting point for their own package. (For example, the [`solarized-one-light-ui`](https://github.com/senki/solarized-one-light-ui) package is a customization of the `one-light-ui` package that ships with Atom.) Some package authors maintain their package as a fork of the core package, and they periodically merge changes (e.g., bug fixes) from the original package into their customized package. The workflow for merging upstream changes into their fork will need to change somewhat following the [package consolidation effort](https://github.com/atom/atom/blob/051bfe5d22d209e8fb4a1d91fe2a9251c97f370c/docs/rfcs/003-consolidate-core-packages.md).

This pull request aims to offer an updated workflow that will hopefully allow the affected package authors to continue incorporating upstream changes into their forks.

### TODO

- [x] Add guide for maintaining an existing fork that was created prior to the package moving into atom/atom
- [x] Add guide for creating a fork for a package that already resides in atom/atom

---

Refs: https://github.com/atom/atom/issues/17856#issuecomment-415701441

jasonrudolph commented:
@lee-dohm: When you have a moment, would you mind reviewing these updates? (This is my first time adding a guide to the flight manual, so I'm probably overlooking some things, and I'm hopeful that you can help me identify any such issues.) In addition to general feedback, what do you think about having this content as a new appendix? Does that seem appropriate, or is there a better home for this content?

/fyi @maxbrunsfeld @daviwil

lee-dohm commented on content/forking-a-core-package/sections/maintaining-a-fork-of-a-core-package-in-atom-atom.md:
```diff
@@ -0,0 +1,69 @@
+
+{{#tip}}
+
+**Tip:** Follow theses steps each time you want to merge upstream changes into your fork.
```
Follow theses &rarr; Follow these

lee-dohm commented on content/forking-a-core-package/sections/creating-a-fork-of-a-core-package-in-atom-atom.md:
```diff
@@ -0,0 +1,48 @@
+
+1. Download the [current contents of the atom/atom repository as a gzipped tarball](https://github.com/atom/atom/archive/master.tar.gz)
+
+2. Unzip the file to a temporary location (for example `/tmp/atom`)
```
You can have separate example paths for Windows and Linux/Mac by using [tagged spans](https://github.com/atom/flight-manual.atom.io/blob/master/CONTRIBUTING.md#os-specific-sections):

```markdown
(for example <span class='platform-mac platform-linux'>`/tmp/atom`</span><span class='platform-windows'>`C:\TEMP\atom`</span>)
```

lee-dohm commented on content/forking-a-core-package/sections/creating-a-fork-of-a-core-package-in-atom-atom.md:
````diff
@@ -0,0 +1,48 @@
+
+    ``` command-line
+    $ cp -R /tmp/atom/packages/one-light-ui ~/src/one-light-ui-plus
+    ```
````
I also tend to avoid ordered lists for instructions because the OS-specific blocks (spans are fine though) don't interact well with ordered lists. And I tend to prefer to have OS-specific blocks around sections like this where the Unix-style and Windows-style commands are very different.

lee-dohm commented:
This looks pretty good! I feel it fits better in "Chapter 3: Hacking Atom", probably right after "Contributing to official Atom packages", than as an appendix though.

Let me know if I can help in any way :+1:

jasonrudolph commented:
@lee-dohm: Thanks for the review. :bow::zap:

I think I've addressed all your feedback.

> I also tend to avoid ordered lists for instructions because the OS-specific blocks (spans are fine though) don't interact well with ordered lists. -- https://github.com/atom/flight-manual.atom.io/pull/484/files/a7bde9a7fc7c8453c7469b87f8c5b14f3c88e742#r215486052

I _think_ I've managed to overcome any issues with OS-specific content and the ordered list. Below are screenshots of the new pages. If you feel strongly that an ordered list is a poor choice here, please let me know, and I'll look for another approach.

## Screenshots

### Creating a fork (Windows)

![screencapture-localhost-4000-hacking-atom-sections-creating-a-fork-of-a-core-package-in-atom-atom-2018-09-06-12_51_22](https://user-images.githubusercontent.com/2988/45172730-3aa6e780-b1d4-11e8-8b69-be4c53e487c8.png)

### Creating a fork (Mac)

![screencapture-localhost-4000-hacking-atom-sections-creating-a-fork-of-a-core-package-in-atom-atom-2018-09-06-12_51_29](https://user-images.githubusercontent.com/2988/45172731-3aa6e780-b1d4-11e8-9238-782e8bf57a26.png)

### Maintaining a fork

![screencapture-localhost-4000-hacking-atom-sections-maintaining-a-fork-of-a-core-package-in-atom-atom-2018-09-06-12_51_44](https://user-images.githubusercontent.com/2988/45172733-3aa6e780-b1d4-11e8-8ac5-1b22fb8d1d08.png)

lee-dohm commented:
Looks great! Thanks so much for the hard work @jasonrudolph :bow:
